### ISSUE TITLE
establish guardrails around what can/can't be public

### ISSUE THREAD

**Issue by afeld** [18F/aws-admin#92]
## Background information

As we are doing more work Terraforming parts of our infrastructure, I want to make sure we are all on the same page (with each other and Security) about what is/isn't ok to be listed publicly. Examples:

- IAM usernames and their permissions
- Account numbers
- Budget values

## Implementation steps

- [x] Follow-up issue created around implementing those rules
- [x] Share with #privacy
- [x] Share with #dev
- [x] Include in the newsletter

## Acceptance criteria

- [x] We have documented what can/can't be public around TTS-wide AWS configuration
- [x] We have consensus within the Tech Portfolio
- [x] We have consensus with Security

---

The assignee can add some checkboxes as a "sketch" of the steps to complete, which may evolve.

**Comment by JJediny** [comment]
- [ ] Review AWS documentation
- [ ] [Factor in our DRAFT government CUI guidance](https://docs.google.com/document/d/1P-_ow-6CFu_uEJ62wLHTUhix-d5zWVls-hdTPD16bho/edit)

**Comment by its-a-lisa-at-work** [comment]
TheIT Security Procedural Guide: Web Application Security CIO-IT Security-07-35 ts the following Key Security Principles that could be helpful guidelines to determine these guardrails

> Key Security Principles
Follow the security principles identified below to employ the proper security precautions. Keep these principles in mind throughout the development and implementation of any system.
Minimize Exposure – Code cannot be exploited if it is not accessible.
Secure Defaults – It is best to start secure, adding functionality as needed.
Principle of Least Privilege – Limiting privileges minimizes the potential damage from attacks.
Defense in Depth – Implement several layers of security.
Fail Safe – When code breaks, ensure security does not.
Privilege Separation – Divide applications into components each limited to the specific privileges required to perform their task(s), limiting potential damage from security attacks.
Simplicity – Complexity increases the probability of security issues.
Input Validation – Always validate input on the server side.
Network Segmentation – Implement appropriate network segmentation to separate web and database tiers and development, test, and staging environments from production.

**Comment by its-a-lisa-at-work** [comment]
Happy to pair on this with someone! I'm particularly interested to see the outcome ya'll have on this and can provide my view point as well.

**Comment by JJediny** [comment]
> cite: Information_Security_Program_Plan_(ISPP)_[CIO_IT_Security_18-90_Rev_3]_06-16-2020docx.pdf

### IR-6 Part B

#### Information Impact:
>Describes the type of information lost, compromised, or corrupted.

**NO IMPACT:** 
No known data impact.

**SUSPECTED BUT NOT IDENTIFIED:** 
A data loss or impact to availability is suspected, but no direct confirmation exists.

**PRIVACY DATA BREACH:** 
The confidentiality of personally identifiable information (PII) or personal health information (PHI) was compromised.

**PROPRIETARY INFORMATION BREACH:** 
The confidentiality of unclassified proprietary information, such as protected critical infrastructure information (PCII), intellectual property, or trade secrets was compromised.

**DESTRUCTION OF NON-CRITICAL SYSTEMS:** 
Destructive techniques, such as master boot record (MBR) overwrite; have been used against a
non-critical system.

**CRITICAL SYSTEMS DATA BREACH:** 
Data pertaining to a critical system has been exfiltrated.

**CORE CREDENTIAL COMPROMISE:** 
Core system credentials (such as domain or enterprise administrative credentials) or credentials for critical systems have been exfiltrated.

**DESTRUCTION OF CRITICAL SYSTEM:** 
Destructive techniques, such as MBR overwrite; have been used against a critical system.

**Comment by afeld** [comment]
Larger question: https://github.com/18F/tts-tech-portfolio/issues/1018

**Comment by afeld** [comment]
Email thread: `Quick Question: list of AWS specific sensitive information?`

**Comment by afeld** [comment]
cc https://github.com/18F/tts-tech-portfolio/issues/1106

**Comment by its-a-lisa-at-work** [comment]
Checked in with IT Security and based on where this is falling within their list of priorities and the limited capacity through the Holidays, was informed that the earliest we'll receive an answer is in 2-3 weeks so plan on linking back with them early 01/21

**Comment by its-a-lisa-at-work** [comment]
Got a response from `GSA IT Security`

Information deemed sensitive:

- Passwords
- API Keys
- Usernames
- Emails
- IP Addresses
- Subnets
- Resource IDs
- Account IDs
- PII
- PCI
- CUI
- FTI
- ePHI/PHI

> In order to manage AWS accounts in our terraform we need to have the Account number available (any concerns with those being public)? 

Yes there are concerns, Account numbers cannot be released publicly

> Are government users (by their @gsa.gov email accounts) ok to have public, along with their associated AWS roles/policies/groups?  

GSA email addresses are not considered sensitive and freely available now.  What roles, policies and groups someone belongs to is not allowed. 

`From the technical side, we recommend teams to put all this information in a private repository,  NOT in a public repository. Password, API keys, private keys etc we ask the teams NOT to keep in repos at all and use alternative secret management approach/solution.`

**Comment by afeld** [comment]
Great! Very clear. Let's get documented in Before You Ship and we can close this out 🎬

**Comment by JJediny** [comment]
Draft for including into the Guides (please comment):
https://docs.google.com/document/d/1P-_ow-6CFu_uEJ62wLHTUhix-d5zWVls-hdTPD16bho/edit#heading=h.ga82ke57vhn2

**Comment by adborden** [comment]
Just to clarify "emails" refers to messages, not email addresses?

**Comment by adborden** [comment]
There's overlap with the Open Source Policy practices. The practices should live separately from the policy, so eventually these practices should be moving into the handbook. For this story, let's remove the sensitive info section from the open source policy practices.

- [ ] Move sensitive information guidance out of the [open source policy practices](https://github.com/18F/open-source-policy/blob/master/practice.md#protecting-sensitive-information)

**Comment by adborden** [comment]
In terms of _implementing_ this guidance, is there any work here to do? I would want to announce the update to TTS programs and have them reach out if there is anything to address. I wouldn't create a new issue for that. Are there any additional implementation issues that we should create before closing this?

**Comment by adborden** [comment]
I shared this with #privacy to request feedback (in slack).

**Comment by adborden** [comment]
- [ ] Let's announce this guidance in #dev and in the newsletter.

**Comment by adborden** [comment]
I included this in the Newsletter, just waiting on @afeld to approve the language and I will use that for an announcement to #dev.

**Comment by adborden** [comment]
Announced in Slack as well.
